### ISSUE TITLE
mAP lower than paper

### ISSUE THREAD

**Issue by xujingtju** [TuSimple/simpledet#78]

**Comment by RogerChern** [comment]
Could you please provide more info about what you are experimenting with?

**Comment by xujingtju** [comment]
> Could you please provide more info about what you are experimenting with?

I do distributed training on 5 machines with 4 gpus on one machine. I use config/tridentnet_r101v2c4_c5_multiscale_addminival_3x_fp16.py with KvstoreParam.gpus  modified  into [0, 1, 2, 3].finally, i get 45.4 mAP on test-dev set.i dont konw if i need modify some parameters with different gpu numbers? thank you

**Comment by xujingtju** [comment]
> Could you please provide more info about what you are experimenting with?

by the way,  when I train the trident net on one machine with 4 gpus using config/tridentnet_r101v2c4_c5_1x.py,i get 41.1 mAP on minival set, a little higher than 40.6 in paper.

**Comment by RogerChern** [comment]
The batch norm is synced within a single machine, so a 4-GPU machine will yield a bn batch size of 12 instead of 24 which will severely harm the performance.

Only `config/tridentnet_r101v2c4_c5_multiscale_addminival_3x_fp16.py` will be affected since it is the only model adopting syncbn. You can try to switch to some 8-GPU machines or you can use fixed bn for training on a 4-GPU machine and turn on `branch_bn_shared` accordingly. This will yield a slightly lower result than a 48.4 mAP.

@xujingtju

**Comment by RogerChern** [comment]
We will switch to horovod for distributed training and rewrite the syncbn for multi-node synchronization in the future.
